src/EventStore.Projections.Core/Services/Processing/ResultEventEmitter.cs

```diff
@@ -20,21 +23,31 @@ public class ResultEventEmitter : IResultEventEmitter {
 			CheckpointTag at) {
 			var streamId = _namesBuilder.MakePartitionResultStreamName(partition);
 			var allResultsStreamId = _namesBuilder.GetResultStreamName();
+			EmittedEventEnvelope result;
 			if (string.IsNullOrEmpty(partition)) {
-				var result =
+				result =
 					new EmittedEventEnvelope(
 						projectionResult == null
 							? new EmittedDataEvent(
 								streamId, Guid.NewGuid(), "ResultRemoved", true, null, null, at, null)
 							: new EmittedDataEvent(
 								streamId, Guid.NewGuid(), "Result", true, projectionResult, null, at, null),
 						_resultStreamMetadata);
+				if (!string.IsNullOrEmpty(result.Event.Data) && !warningProjectionList.Contains(_namesBuilder.EffectiveProjectionName)) {
```

Comment:
this will do a linear search through the list every time a projection emits

Reply:
I have pushed new changes as per which the list would only be searched when an event of event type "Result" is emitted, and not for any other types of emitted events.